File: Szlaki/Models/Trail.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Diagnostics.CodeAnalysis;
using System.IO;
using System.Linq;
using System.Threading.Tasks;

namespace Szlaki.Models
{
    public class Trail
    {
        [Key]
        public string Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        [ForeignKey("UserId")]
        public ApplicationUser User { get; set; }
        public string UserId { get; set; }
        public virtual List<Photo> Photos { get; set; }
        public virtual List<Video> Videos { get; set; }

    }
}

File: Szlaki/ClientApp/src/app/modules/trails/models/trail.ts
import { Photos } from './photoModel';
import { from } from "rxjs";

export class Trail {
    id: string;
    title: string;
    description: string;
    startDate: Date;
    endDate: Date;
    photos: any[];
    videos: any[];
    user: any;
    userId: any;
}